contracts/augur/trading/ZeroXTrade.sol

```diff
@@ -191,26 +48,31 @@ contract ZeroXTrade is Initializable, IZeroXTrade, IERC1155 {
         address _affiliateAddress,
         bytes32 _tradeGroupId,
         IExchange.Order[] memory _orders,
-        bytes[] memory _signatures
+        bytes[] memory _signatures,
+        address _taker
     )
         public
         payable
         returns (uint256)
     {
         uint256 _fillAmountRemaining = _requestedFillAmount;
 
-        transferFromAllowed = true;
+        // transferFromAllowed = true;
 
-        uint256 _protocolFee = 150000 * tx.gasprice;
+        // uint256 _protocolFee = 150000 * tx.gasprice;
 
         // Do the actual asset exchanges
         for (uint256 i = 0; i < _orders.length && _fillAmountRemaining != 0; i++) {
             IExchange.Order memory _order = _orders[i];
-            validateOrder(_order);
-            IExchange _exchange = getExchangeFromAssetData(_order.makerAssetData);
+            // @discuss not sure what validations are required for orders that were signed for Matic and how to retrieve _exchange
```

Comment:
Let me know, what validations are required.

Reply:
I don't understand why we removed the existing logic for retrieving the exchange form the asset data. Why would this differ for Matic?